pykusto/retriever.py

```diff
@@ -0,0 +1,77 @@
+from concurrent.futures import Future, ThreadPoolExecutor, wait
+from itertools import chain
+from threading import Lock
+from typing import Union, Dict, Any, Iterable
+
+POOL = ThreadPoolExecutor(max_workers=1)
+
+
+class Retriever:
+    _retrieve_by_default: bool
+    _items: Union[None, Dict[str, Any]]
+    _future: Union[None, Future]
+    _lock: Lock
+
+    def __new__(cls, *args, **kwargs):
+        if cls is 'Retriever':
+            raise TypeError("Retriever is abstract")
```

Comment:
My mistake. This can be done with ABC as long as there is at least one abstract method. We have other abstract classes that don't have abstract methods, and for those classes we used the solution of overriding `__new__`. Here we have an abstract method, do we can use ABC. Fixed, and also added comments in the other classes to make it clear why ABC wasn't used.